src/mesh/mesh_adaptation.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mesh_adaptation.h"
+
+namespace PHiLiP {
+
+template <int dim, typename real, typename MeshType>
+MeshAdaptation<dim,real,MeshType>::MeshAdaptation(double critical_res_input, int total_ref_cycle, double refine_frac, double coarsen_frac)
+    : critical_residual(critical_res_input)
+    , total_refinement_cycles(total_ref_cycle)
+    , refinement_fraction(refine_frac)
+    , coarsening_fraction(coarsen_frac)
+    , current_refinement_cycle(0)
+    {}
+
+template <int dim, typename real, typename MeshType>
+int MeshAdaptation<dim,real,MeshType>::adapt_mesh(std::shared_ptr< DGBase<dim, real, MeshType> > dg)
```

Comment:
Instead of returning 0s, make the function void!

Reply:
Same applies for the other functions. If you don't use the return value, there is no point in returning it.

Reply:
Sure, fixed it.